src/campaign.cairo

```diff
@@ -142,20 +195,38 @@ mod TokengiverCampaign {
             self.withdrawal_balance.write(campaign_address, amount);
         }
 
+        fn withdraw(ref self: ContractState, campaign_address: ContractAddress, amount: u256) {
+            let campaign: Campaign = self.campaign.read(campaign_address);
+            let caller: ContractAddress = get_caller_address();
+
+            assert(caller == campaign.campaign_owner, NOT_CAMPAIGN_OWNER);
+
+            let available_balance: u256 = self.withdrawal_balance.read(campaign_address);
+            assert(amount <= available_balance, INSUFFICIENT_BALANCE);
+
+            self.withdrawal_balance.write(campaign_address, available_balance - amount);
```

Comment:
this should go after transfer, only run this when the transaction , 

check to ensure the transfer was successful via ASSERTION, once it is true,

`self.withdrawal_balance.write(campaign_address, available_balance - amount);`